protoc.Dockerfile

```diff
@@ -0,0 +1,16 @@
+# Docker image for protoc
+ARG BASE="alpine:3.14.3"
+FROM $BASE
+ARG PROTOC_VERSION="3.19.1"
+ARG BIN=""
+
+RUN apk add gcompat
+ADD "https://github.com/protocolbuffers/protobuf/releases/download/v$PROTOC_VERSION/protoc-$PROTOC_VERSION-linux-x86_64.zip" protoc.zip
+RUN mkdir /usr/local/lib/protoc && unzip protoc.zip -d /usr/local/lib/protoc && rm protoc.zip
```

Comment:
FYI, your `rm` here doesn't actually clear up any space, since the file data will remain in the storage layer of the `ADD` command. You could solve that by creating and removing the file in the same RUN statement, e.g.

```
RUN wget -o protoc.zip https://... && unzip protoc.zip && rm protoc.zip
```

Or perhaps even

```
RUN wget -o - https://... | unzip -
```

Or something like that. You may not have wget on alpine, but you may have curl or `busybox wget` or something like that :)

Reply:
Right, thanks for that tip! I made the Dockerfile such that the base image can be replaced externally, so then the other base images will also need wget or curl. For simplicity we could keep it the way it is, unless you think the size of the image could be a concern.